src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -14591,5 +14575,199 @@ static int Test1(out int x)
 
             Assert.Equal("System.Int32", model.GetTypeInfo(yRef).Type.ToTestDisplayString());
         }
+
+        [Fact, WorkItem(13219, "https://github.com/dotnet/roslyn/issues/13219")]
+        public void IndexingDynamic()
```

Comment:
>IndexingDynamic [](start = 20, length = 15)

Please add test for scenario that uses ```var``` as type.
Please add verification for SemanticModel behavior.
Please add specific test for type inference before the indexing is bound.